src/core/include/mp-units/framework/value_cast.h

```diff
@@ -152,4 +178,106 @@ template<Unit auto ToU, Representation ToRep, typename QP>
                         std::remove_reference_t<QP>::point_origin};
 }
 
+/**
+ * @brief Explicit cast of a quantity point's representation
+ *
+ * Implicit conversions between quantities of different types are allowed only for "safe"
+ * (e.g. non-truncating) conversion. In truncating cases an explicit cast have to be used.
+ *
+ * inline constexpr struct A : absolute_point_origin<A, isq::distance> A;
+ * inline constexpr struct B : relative_point_origin<A + 1*m> B;
+ *
+ * using ToQP = quantity_point<mm, B, int>;
+ * auto qp = value_cast<ToQP>(quantity_point{1.23 * m});
```

Comment:
> Anyway, this example needs to be fixed

Right, indeed obvious copy-paste error.

> I am not sure if the `value_cast<Q>(qp)` is really needed?

`value_cast<Q>(qp)` operations are convenient as part of implementations of `value_cast<QP>(qp)`-like operations (see below). While my implementation of the latter tries to select the best possible implementation based on the provided types, an expert user may know more about the value range that is in fact being used in their case and instead want to implement a different order of operations. Then, they would also like to be able to use `value_cast<Q>(qp)`. Finally, it provides some symmetry to `value_cast<Q>(q)`. On the other hand, one could argue that `value_cast<Q>(qp)` misleadingly suggests that the result type is going to be `Q`, which of course it is not. However, that same argument could be made for the `value_cast<Repr>(...)` overloads. Anyway, I don't have all too strong feelings here, we can also drop it.

Reply:
I do not think it is needed and it may be misleading indeed.

`value_cast<Q>(q)` makes sense as someone may have a typedef for `quantity` and does not want to extract a unit and rep all the time for a cast. The same applies to `quantity_point`.

On the other hand, `value_cast<Q>(qp)` is strange, as one should not have a `quantity` typedef for an abstraction modeled through a `quantity_point`.

Reply:
> Writing value_cast<typename ToQP::quantity_type>(qp) is much shorter than quantity_point{value_cast<typename ToQP::quantity_type>(qp.quantity_from(FromQP::point_origin)), FromQP::point_origin} and also safer

Why not to write: `quantity_point{ToQP::unit, typename ToQP::rep>(qp}`? I know that it is still spelling two things instead of one but I would like to avoid suggesting people that they should make aliases for `quantity_point` and `quantity` for the same abstraction to make it work. Not everyone will be smart enough or restricting themselves to just use `typename QP::quantity_type` as you mentioned above.

But if you think it is the only way to make it work, then please commit it to the repo. I just don't know how to get feedback about it before putting this under standardization. We will not be able to easily verify how many people depend on this feature in their code. Probably for ISO I will not propose it in the first run, and add it later if people will request for it.

Reply:
OK, let's try it then 😃